src/main/org/pantsbuild/jarjar/util/StandaloneJarProcessor.java

```diff
@@ -52,7 +52,7 @@ public static void run(File from, File to, JarProcessor proc) throws IOException
                     } else if (struct.name.endsWith("/")) {
                         // TODO(chrisn): log
                     } else {
-                        throw new IllegalArgumentException("Duplicate jar entries: " + struct.name);
+                        throw new DuplicateJarEntryException(from.getAbsolutePath(), struct.name);
```

Comment:
Yeah I updated this patch to change the behavior of that block; it now prints the error and re-throws it.